speed-dreams/source-2.2.3/src/drivers/replaydriver/Driver.cpp

```diff
@@ -0,0 +1,180 @@
+/// @param p_situation The current race situation
+void Driver::Drive(tCarElt* p_car, tSituation* p_situation)
+{
+    std::string accelString;
+    m_replayFile >> accelString;
+    float accel = std::stof(accelString);
+    std::string brakeString;
+    m_replayFile >> brakeString;
+    float brake = std::stof(brakeString);
+    std::string steerString;
+    m_replayFile >> steerString;
+    float steer = std::stof(steerString);
+    std::string gearString;
+    m_replayFile >> gearString;
+    float gear = std::stof(gearString);
+    std::string clutchString;
+    m_replayFile >> clutchString;
+    float clutch = std::stof(clutchString);
+
+    std::string raceCmdString;
+    m_replayFile >> raceCmdString;
+    float raceCmd = std::stof(raceCmdString);
+    std::string lightCmdString;
+    m_replayFile >> lightCmdString;
+    float lightCmd = std::stof(lightCmdString);
+    std::string ebrakeCmdString;
+    m_replayFile >> ebrakeCmdString;
+    float ebrakeCmd = std::stof(ebrakeCmdString);
+    std::string brakeFLCmdString;
+    m_replayFile >> brakeFLCmdString;
+    float brakeFLCmd = std::stof(brakeFLCmdString);
+    std::string brakeFRCmdString;
+    m_replayFile >> brakeFRCmdString;
+    float brakeFRCmd = std::stof(brakeFRCmdString);
+    std::string brakeRLCmdString;
+    m_replayFile >> brakeRLCmdString;
+    float brakeRLCmd = std::stof(brakeRLCmdString);
+    std::string brakeRRCmdString;
+    m_replayFile >> brakeRRCmdString;
+    float brakeRRCmd = std::stof(brakeRRCmdString);
+    std::string wingFCmdString;
+    m_replayFile >> wingFCmdString;
+    float wingFCmd = std::stof(wingFCmdString);
+    std::string wingRCmdString;
+    m_replayFile >> wingRCmdString;
+    float wingRCmd = std::stof(wingRCmdString);
+    std::string telemetryModeString;
+    m_replayFile >> telemetryModeString;
+    float telemetryMode = std::stof(telemetryModeString);
+    std::string singleWheelBrakeModeString;
+    m_replayFile >> singleWheelBrakeModeString;
+    float singleWheelBrakeMode = std::stof(singleWheelBrakeModeString);
+
+    p_car->_accelCmd = accel;
+    p_car->_brakeCmd = brake;
+    p_car->_steerCmd = steer;
+    p_car->_gearCmd = static_cast<int>(gear);
+    p_car->_clutchCmd = clutch;
+    p_car->_raceCmd = static_cast<int>(raceCmd);
+    p_car->_lightCmd = static_cast<int>(lightCmd);
+    p_car->_ebrakeCmd = static_cast<int>(ebrakeCmd);
+    p_car->_brakeFLCmd = brakeFLCmd;
+    p_car->_brakeFRCmd = brakeFRCmd;
+    p_car->_brakeRLCmd = brakeRLCmd;
+    p_car->_brakeRRCmd = brakeRRCmd;
+    p_car->_wingFCmd = wingFCmd;
+    p_car->_wingRCmd = wingRCmd;
+    p_car->_telemetryMode = static_cast<int>(telemetryMode);
+    p_car->_singleWheelBrakeMode = static_cast<int>(singleWheelBrakeMode);
+    std::string inputTime;
+    m_replayFile >> inputTime;
+
+    if (m_replayFile.eof())
+    {
+        p_situation->raceInfo.state = RM_RACE_ENDED;
+    }
+    else
+    {
+        m_inputTime = std::stod(inputTime);
+    }
```

Comment:
Shouldn't you check for EOF before reading the new input time?
```suggestion
    if (m_replayFile.eof())
    {
        p_situation->raceInfo.state = RM_RACE_ENDED;
    }
    else
    {
        std::string inputTime;
        m_replayFile >> inputTime;
        m_inputTime = std::stod(inputTime);
    }
```